parquet-common/src/main/java/org/apache/parquet/SemanticVersion.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Very basic semver parser, only pays attention to major, minor, and patch numbers.
+ * Attempts to do a little bit of validation that the version string is valid, but
+ * is not a full implementation of the semver spec.
+ *
+ * NOTE: compareTo only respects major, minor, and patch (ignores rc numbers, SNAPSHOT, etc)
+ */
+public final class SemanticVersion implements Comparable<SemanticVersion> {
+  // (major).(minor).(patch)[(rc)(rcnum)]?(-(SNAPSHOT))?
+  private static final String FORMAT = "^(\\d+)\\.(\\d+)\\.(\\d+)((.*)(\\d+))?(\\-(.*))?$";
+  private static final Pattern PATTERN = Pattern.compile(FORMAT);
+
+  public final int major;
+  public final int minor;
+  public final int patch;
+
+  public SemanticVersion(int major, int minor, int patch) {
+    Preconditions.checkArgument(major >= 0, "major must be >= 0");
+    Preconditions.checkArgument(minor >= 0, "minor must be >= 0");
+    Preconditions.checkArgument(patch >= 0, "patch must be >= 0");
+
+    this.major = major;
+    this.minor = minor;
+    this.patch = patch;
+  }
+
+  public static SemanticVersion parse(String version) {
+    Matcher matcher = PATTERN.matcher(version);
+
+    Preconditions.checkArgument(matcher.matches(), "" + version + " does not match format " + FORMAT);
```

Comment:
This method should never throw an Exception. Instead of throwing an ArgumentException if we don't recognize the version, I think this should simply return null: no information about the version is available. Otherwise, this would prevent Parquet MR from reading files written without a compliant version string, even if they are valid according to the format spec. The python implementation and Impala are likely to have problems there.